src/app/_models/modal-dialog.ts

```diff
@@ -0,0 +1,7 @@
+import { Observable } from 'rxjs';
+
+export interface ModalDialog {
```

Comment:
The interface allows circular references to be avoided in the imports: the service manages components, which in turn reference the service